### ISSUE TITLE
[BUG] 0.40.0 PECL install fails on Alpine

### ISSUE THREAD

**Issue by fbnfgc** [DataDog/dd-trace-php#778]
**Install fails**

Using PECL on Alpine linux, installation fails with error message : `/tmp/pear/temp/datadog_trace/src/ext/ddtrace.h:3:10: fatal error: dogstatsd_client/client.h: No such file or directory`.

Steps to reproduce : 
```bash
docker run -it --rm php:7.4-fpm-alpine /bin/sh 
apk add --no-cache autoconf automake file gcc g++ libc-dev make libcurl curl-dev
pecl install datadog_trace
```

Here is the full error (looks like packaging problem ?) : 

```bash
downloading datadog_trace-0.40.0.tgz ...
Starting to download datadog_trace-0.40.0.tgz (192,771 bytes)
.........................................done: 192,771 bytes
62 source files, building
running: phpize
Configuring for:
PHP Api Version:         20190902
Zend Module Api No:      20190902
Zend Extension Api No:   320190902
Enable internal debugging in ddtrace [no] : 

Warning: Use of undefined constant name - assumed 'name' (this will throw an Error in a future version of PHP) in PEAR/Builder.php on line 407

Warning: Use of undefined constant name - assumed 'name' (this will throw an Error in a future version of PHP) in /usr/local/lib/php/PEAR/Builder.php on line 407
building in /tmp/pear/temp/pear-build-defaultuserfdGDlb/datadog_trace-0.40.0
running: /tmp/pear/temp/datadog_trace/configure --with-php-config=/usr/local/bin/php-config --enable-ddtrace-debug=no
configure: WARNING: unrecognized options: --enable-ddtrace-debug
checking for grep that handles long lines and -e... /bin/grep
checking for egrep... /bin/grep -E
checking for a sed that does not truncate output... /bin/sed
checking for pkg-config... /usr/bin/pkg-config
checking pkg-config is at least version 0.9.0... yes
checking for cc... cc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables...
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether cc accepts -g... yes
checking for cc option to accept ISO C89... none needed
checking how to run the C preprocessor... cc -E
checking for icc... no
checking for suncc... no
checking for system library directory... lib
checking if compiler supports -R... no
checking if compiler supports -Wl,-rpath,... yes
checking build system type... x86_64-pc-linux-musl
checking host system type... x86_64-pc-linux-musl
checking target system type... x86_64-pc-linux-musl
checking for PHP prefix... /usr/local
checking for PHP includes... -I/usr/local/include/php -I/usr/local/include/php/main -I/usr/local/include/php/TSRM -I/usr/local/include/php/Zend -I/usr/local/include/php/ext -I/usr/local/include/php/ext/date/lib
checking for PHP extension directory... /usr/local/lib/php/extensions/no-debug-non-zts-20190902
checking for PHP installed headers prefix... /usr/local/include/php
checking if debug is enabled... no
checking if zts is enabled... no
checking for gawk... no
checking for nawk... no
checking for awk... awk
checking if awk is broken... no
checking whether to enable Datadog tracing support... yes, shared
checking whether to enable AddressSanitizer for ddtrace... no
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking execinfo.h usability... no
checking execinfo.h presence... no
checking for execinfo.h... no
checking for library containing backtrace... no
checking for shm_open in -lrt... yes
checking for curl_easy_setopt in -lcurl... yes
checking time.h usability... yes
checking time.h presence... yes
checking for time.h... yes
checking for a sed that does not truncate output... /bin/sed
checking for ld used by cc... /usr/x86_64-alpine-linux-musl/bin/ld
checking if the linker (/usr/x86_64-alpine-linux-musl/bin/ld) is GNU ld... yes
checking for /usr/x86_64-alpine-linux-musl/bin/ld option to reload object files... -r
checking for BSD-compatible nm... /usr/bin/nm -B
checking whether ln -s works... yes
checking how to recognize dependent libraries... pass_all
checking dlfcn.h usability... yes
checking dlfcn.h presence... yes
checking for dlfcn.h... yes
checking the maximum length of command line arguments... 98304
checking command to parse /usr/bin/nm -B output from cc object... ok
checking for objdir... .libs
checking for ar... ar
checking for ranlib... ranlib
checking for strip... strip
checking if cc supports -fno-rtti -fno-exceptions... no
checking for cc option to produce PIC... -fPIC
checking if cc PIC flag -fPIC works... yes
checking if cc static flag -static works... yes
checking if cc supports -c -o file.o... yes
checking whether the cc linker (/usr/x86_64-alpine-linux-musl/bin/ld -m elf_x86_64) supports shared libraries... yes
checking whether -lc should be explicitly linked in... no
checking dynamic linker characteristics... GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... no

creating libtool
appending configuration tag "CXX" to libtool
configure: patching config.h.in
configure: creating ./config.status
config.status: creating config.h
configure: WARNING: unrecognized options: --enable-ddtrace-debug
running: make
/bin/sh /tmp/pear/temp/pear-build-defaultuserfdGDlb/datadog_trace-0.40.0/libtool --mode=compile cc -DZEND_ENABLE_STATIC_TSRMLS_CACHE=1 -Wall -std=gnu11 -I. -I/tmp/pear/temp/datadog_trace -DPHP_ATOM_INC -I/tmp/pear/temp/pear-build-defaultuserfdGDlb/datadog_trace-0.40.0/include -I/tmp/pear/temp/pear-build-defaultuserfdGDlb/datadog_trace-0.40.0/main -I/tmp/pear/temp/datadog_trace -I/usr/local/include/php -I/usr/local/include/php/main -I/usr/local/include/php/TSRM -I/usr/local/include/php/Zend -I/usr/local/include/php/ext -I/usr/local/include/php/ext/date/lib -I/tmp/pear/temp/pear-build-defaultuserfdGDlb/datadog_trace-0.40.0/. -I/tmp/pear/temp/pear-build-defaultuserfdGDlb/datadog_trace-0.40.0/src/ext -I/tmp/pear/temp/pear-build-defaultuserfdGDlb/datadog_trace-0.40.0/src/ext/mpack -I/tmp/pear/temp/pear-build-defaultuserfdGDlb/datadog_trace-0.40.0/src/dogstatsd  -DHAVE_CONFIG_H  -g -O2   -c /tmp/pear/temp/datadog_trace/src/ext/ddtrace.c -o src/ext/ddtrace.lo
mkdir src/ext/.libs
 cc -DZEND_ENABLE_STATIC_TSRMLS_CACHE=1 -Wall -std=gnu11 -I. -I/tmp/pear/temp/datadog_trace -DPHP_ATOM_INC -I/tmp/pear/temp/pear-build-defaultuserfdGDlb/datadog_trace-0.40.0/include -I/tmp/pear/temp/pear-build-defaultuserfdGDlb/datadog_trace-0.40.0/main -I/tmp/pear/temp/datadog_trace -I/usr/local/include/php -I/usr/local/include/php/main -I/usr/local/include/php/TSRM -I/usr/local/include/php/Zend -I/usr/local/include/php/ext -I/usr/local/include/php/ext/date/lib -I/tmp/pear/temp/pear-build-defaultuserfdGDlb/datadog_trace-0.40.0/. -I/tmp/pear/temp/pear-build-defaultuserfdGDlb/datadog_trace-0.40.0/src/ext -I/tmp/pear/temp/pear-build-defaultuserfdGDlb/datadog_trace-0.40.0/src/ext/mpack -I/tmp/pear/temp/pear-build-defaultuserfdGDlb/datadog_trace-0.40.0/src/dogstatsd -DHAVE_CONFIG_H -g -O2 -c /tmp/pear/temp/datadog_trace/src/ext/ddtrace.c  -fPIC -DPIC -o src/ext/.libs/ddtrace.o
In file included from /tmp/pear/temp/datadog_trace/src/ext/ddtrace.c:25:
/tmp/pear/temp/datadog_trace/src/ext/ddtrace.h:3:10: fatal error: dogstatsd_client/client.h: No such file or directory
    3 | #include <dogstatsd_client/client.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make: *** [Makefile:192: src/ext/ddtrace.lo] Error 1
ERROR: `make' failed
```

**PHP Info**
- Output of `php -v`
```bash
PHP 7.4.3 (cli) (built: Feb 20 2020 21:53:46) ( NTS )
Copyright (c) The PHP Group
Zend Engine v3.4.0, Copyright (c) Zend Technologies
```

- Output of `php -m`
```bash
[PHP Modules]
Core
ctype
curl
date
dom
fileinfo
filter
ftp
hash
iconv
json
libxml
mbstring
mysqlnd
openssl
pcre
PDO
pdo_sqlite
Phar
posix
readline
Reflection
session
SimpleXML
sodium
SPL
sqlite3
standard
tokenizer
xml
xmlreader
xmlwriter
zlib

[Zend Modules]
```
**OS Info**
```bash
NAME="Alpine Linux"
VERSION_ID=3.11.3
PRETTY_NAME="Alpine Linux v3.11"

```
**Output of dd-doctor**
```bash
WARNING: Script is running from the CLI SAPI.
         Please run this script from your web browser.

DataDog trace extension verification

- PHP version and SAPI                               [7.4.3 - cli]
- ddtrace extension installed                        [FAIL]
- ddtrace version (installed)                        []
- ddtrace version (const)                            []
- ddtrace version (userland)                         []
- ddtrace versions in sync                           [OK]
- dd_trace() function available                      [FAIL]
- dd_trace_env_config() function available           [FAIL]
- ddtrace.request_init_hook set                      [FAIL]
- ddtrace.request_init_hook reachable                [FAIL]
- open_basedir INI directive                         [empty]
- 'open_basedir' allows access to '/proc/self/'      [OK]
- IntegrationsLoader exists                          [FAIL]
- DDTrace\Tracer class exists                        [FAIL]
- Background sender is enabled?                      [NO]
  > You can enable the background sender via
    DD_TRACE_BETA_SEND_TRACES_VIA_THREAD=true
- Configured Agent host                              [localhost]
- Configured Agent port                              [8126]
- Agent can receive traces                           [FAIL]
Curl verbose output: 

*   Trying 127.0.0.1:8126...
* TCP_NODELAY set
* connect to 127.0.0.1 port 8126 failed: Connection refused
*   Trying ::1:8126...
* TCP_NODELAY set
* Immediate connect fail for ::1: Address not available
*   Trying ::1:8126...
* TCP_NODELAY set
* Immediate connect fail for ::1: Address not available
* Failed to connect to localhost port 8126: Connection refused
* Closing connection 0
```

**Comment by morrisonlevi** [comment]
Thanks for the report. It does look like it is a packaging issue.

**Comment by morrisonlevi** [comment]
Pecl installs should work in 0.41.1. Please read the description of this PR as it has information about the limits of pecl installs: https://github.com/DataDog/dd-trace-php/pull/781. We may improve this in the future, but this is unscheduled at this time.

**Comment by fbnfgc** [comment]
Thanks!

**Comment by axot** [comment]
One question about installing ddtrace via pecl, does it need be same ddtrace version between php composer's and pecl's?

**Comment by labbati** [comment]
Answered here: https://github.com/DataDog/dd-trace-php/issues/1115#issuecomment-741832980

TL;DR it is recommended, otherwise you might use from composer methods that are not actually available in the extension.
